### DISCUSSION TITLE
Protect additional routes with SWS authentication

### DISCUSSION THREAD

**Discussion by NathanSavageKaimai** [slanatech/swagger-stats#278] [Ideas]
Hi All,

Im loving using swagger stats as the admin panel for my app. I just had a quick idea that could make it even more useful.
Swagger Stats is the core of my admin panel however there are also other panels I am using such as bull board (https://github.com/felixmosh/bull-board/tree/master) to monitor my job queues.

I think it would be really useful if I could extend the authentication middleware to additional routes for these other panels so that it can all be handled in one place. Also, if in the swagger stats header bar, I could place links to my additional panels?

Let me know what you think, If people like the idea i can make a branch for it.
I have a feeling that it may be possible to bodge in the route protection with no package changes already however I am not well versed in express middleware so if anyone has any ideas on how to do this then I would love to hear them!

**Comment by sv2** [comment]
Sounds like a good idea, thanks !
